sequencesender/sequencesender.go

```diff
@@ -96,19 +100,58 @@ func (s *SequenceSender) tryToSendSequence(ctx context.Context) {
 			mTxResultLogger := ethtxmanager.CreateMonitoredTxResultLogger(ethTxManagerOwner, result)
 			mTxResultLogger.Error("failed to send sequence, TODO: review this fatal and define what to do in this case")
 		}
+
+		if len(result.Txs) == 1 {
+			var txL1BlockNumber uint64
+			var txHash common.Hash
+			for _, tx := range result.Txs {
+				txL1BlockNumber = tx.Receipt.BlockNumber.Uint64()
+				txHash = tx.Tx.Hash()
+				break
+			}
```

Comment:
I think this code should be like this:
```go
if len(result.Txs) > 0 {
    var txL1BlockNumber uint64
    var txHash common.Hash
    for _, tx := range result.Txs {
        if tx.Receipt != nil {
	        txL1BlockNumber = tx.Receipt.BlockNumber.Uint64()
	        txHash = tx.Tx.Hash()
        }
    }
```

The monitored TX can have multiple TXs. Also, only one of the TXs will have a receipt; the others were discarded in the pool because a new one with a better fee was mined.

Reply:
Ok, I misunderstand the Txs field, I thought this was a map with the monitored tx for the sequencer-sender, but I have reviewed the code and the ProcessPendingMonitoredTxs is called for each monitored tx of the owner and Txs is the list of L1 txs retries

Reply:
I have added a "break" when a tx with receipt is found (it should have only one tx with receipt). Also I have put the code that waits for L1 block confirmations inside a "else" block, since that code must be only executed if the monitored tx is confirmed (not failed). Before the code was executed always, also when the tx had failed status. Finally I have added a fatal in case the monitored tx is confirmed and doesn't have any tx with receipt (that shouldn't happen). Please review the update PR

Reply:
I have just changed the if..else order. The "if" block is executed when the tx is confirmed the "else" block when it's failed. I think is more "clean" to do in this way now